core/Test/Tasty/Ingredients/ConsoleReporter.hs

```diff
@@ -128,11 +129,32 @@ buildTestOutput opts tree =
       level <- ask
 
       let
+        postNamePadding = alignment - indentSize * level - stringWidth name
+
+        testNamePadded = printf "%s%s: %s"
+          (indent level)
+          name
+          (replicate postNamePadding ' ')
+
+        resultPosition = length testNamePadded
+
         printTestName = do
-          printf "%s%s: %s" (indent level) name
-            (replicate (alignment - indentSize * level - stringWidth name) ' ')
+          putStr testNamePadded
           hFlush stdout
 
+        printTestProgress progress
+          | progress == emptyProgress = pure ()
+
+          | otherwise = do
+              let
+                msg = case (cleanupProgressText $ progressText progress, 100 * progressPercent progress) of
+                        ("",  pct) -> printf "%.0f%%" pct
+                        (txt, 0.0) -> printf "%s" txt
+                        (txt, pct) -> printf "%s: %.0f%%" txt pct
+              setCursorColumn resultPosition
```

Comment:
This is not enough: if a previous progress text was longer than current, you need to cover up outstanding characters.

Reply:
`txt` is handled by the provider, i.e. `quickcheck` has a logic for exactly that; but indeed we should include extra space for the percent progress.

Reply:
But this will be only necessary if it's a regress, e.g. from `100` to `99`.

Reply:
`quickcheck` is not the only test provider to use progress reporting. I would not expect a test provider to track the length of the previous progress text and pad it with spaces - after all it does not really have an idea how we intend to render it. 

I made a quick attempt for `tasty-bench` to support progress reporting: https://github.com/Bodigrim/tasty-bench/tree/tasty-progress-reporting. Could you please check that `cabal run tasty-bench:benchmarks` renders correctly? At the moment it is mangled, like "iterationsns".

Reply:
It doesn't build with tasty in this branch.

Reply:
It wasn't just that, I also had compilation errors.

Reply:
That's no longer true with the new tip of this branch.